theano/tensor/opt.py

```diff
@@ -2591,6 +2591,23 @@ def local_adv_sub1_adv_inc_sub1(node):
     return [T.cast(y, node.outputs[0].dtype)]
 
 
+@register_specialize
+@register_stabilize
+@register_canonicalize
+@gof.local_optimizer([IncSubtensor, AdvancedIncSubtensor1])
```

Comment:
I think this could also work for AdvancedIncSubtensor. Do you want to try it at the same time? It would just need a test.